### PR TITLE
Change loops conditions to make zero loop risk more obvious.

### PR DESCRIPTION
**Pull request by faramir-dev**
Fixes openssl#18073.

I compiled `libcrypto.so` with flags: `linux-x86_64`, `no-asm` and generated ASM for the modified functions looks to be the same with and without these changes. And the changes increases readability a bit.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/modes/gcm128.c [hunk 1]
```diff
@@ -537,7 +537,7 @@ static void gcm_ghash_4bit(u64 Xi[2], const u128 Htable[16],
             Xi[0] = Z.hi;
             Xi[1] = Z.lo;
         }
-    } while (inp += 16, len -= 16);
+    } while (inp += 16, (len -= 16)>0);
```
Comment (bernd-edlinger) [inline crypto/modes/gcm128.c:540]: but len is size_t so this is just more ugly IMHO

Comment (faramir-dev) [reply crypto/modes/gcm128.c]: I agree, I changed the condition but I am afraid it is still more ugly than a simple comment that `len % 16 == 0`? What do you think?

File: crypto/aria/aria.c [hunk 2]
```diff
@@ -498,7 +498,7 @@ void ossl_aria_encrypt(const unsigned char *in, unsigned char *out,
     ARIA_ADD_ROUND_KEY(rk, reg0, reg1, reg2, reg3);
     rk++;
 
-    while (Nr -= 2) {
+    while ((Nr -= 2)>0) {
```
Comment (t8m) [inline crypto/aria/aria.c:501]: Please add spaces around `>`

File: crypto/modes/gcm128.c [hunk 3]
```diff
@@ -537,7 +537,7 @@ static void gcm_ghash_4bit(u64 Xi[2], const u128 Htable[16],
             Xi[0] = Z.hi;
             Xi[1] = Z.lo;
         }
-    } while (inp += 16, len -= 16);
+    } while (inp += 16, (len >= 16) && (len -= 16));
```
Comment (paulidale) [inline crypto/modes/gcm128.c:540]: Can't say that I'm a fan of this expression.  The best I can come up with is:
```c
        inp += 16;
        len -= 16;
    } while (len > 0);
```
which isn't a great stopping criteria (but just as good as the original).

**Review by paulidale — CHANGES_REQUESTED** [review]

File: crypto/modes/gcm128.c (hunk 1 above)
Comment (t8m) [reply crypto/modes/gcm128.c]: Alternatively we could ossl_assert that len % 16 == 0.

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Review by tmshort — APPROVED** [review]

**Comment by paulidale** [comment]
Merged.

**Pull request closed without merging**